From f10ee9d5eb3a9e306040e7af7596aa63f59b1cac Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@lucaweiss.eu>
Date: Tue, 12 Nov 2024 22:32:15 +0100
Subject: [PATCH 2/2] Link against libintl

As required when compiling on Alpine Linux:

  /usr/lib/gcc/x86_64-alpine-linux-musl/14.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: app/CMakeFiles/lomiri-clock-app.dir/main.cpp.o: undefined reference to symbol 'libintl_bind_textdomain_codeset'
  /usr/lib/gcc/x86_64-alpine-linux-musl/14.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: /usr/lib/libintl.so.8: error adding symbols: DSO missing from command line
  collect2: error: ld returned 1 exit status
---
 CMakeLists.txt     | 1 +
 app/CMakeLists.txt | 1 +
 2 files changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9bc32565..0e598674 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,7 @@ find_package(Qt5Core)
 find_package(Qt5Qml)
 find_package(Qt5Quick)
 find_package(Qt5DBus)
+find_package(Intl REQUIRED)
 
 include(FindGettext)
 if(NOT GETTEXT_FOUND)
diff --git a/app/CMakeLists.txt b/app/CMakeLists.txt
index cfa2ccee..f0e0022d 100644
--- a/app/CMakeLists.txt
+++ b/app/CMakeLists.txt
@@ -23,6 +23,7 @@ target_link_libraries(${APP_HARDCODE} PUBLIC
   Qt5::Core
   Qt5::Qml
   Qt5::Quick
+  ${Intl_LIBRARIES}
 )
 
 install(TARGETS ${APP_HARDCODE} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
-- 
2.47.0

